src/main/java/io/bittiger/ads/activity/AdsSelect.java

```diff
@@ -0,0 +1,65 @@
+package io.bittiger.ads.activity;
+
+import io.bittiger.ads.model.Ad;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+import java.lang.String;
+
+public class AdsSelect {
+    private static AdsSelect instance = null;
+    
+    protected AdsSelect() {
+    }
+
+    public static AdsSelect getInstance() {
+        if (instance == null) {
+            instance = new AdsSelect();
+        }
+        return instance;
+    }
+
+    public List<Ad> getMatchedAds(String[] keywords) {
+        List<Ad> ads = new ArrayList<Ad>();
+        if (keywords == null || keywords.length == 0) {
+            return ads;
+        }
+        
+        try {
+            MemcachedClient cache = AdsDao.getCache();

```

Comment:
And I think all the DB logic should be **invisible** here. 
You'd better create a method in AdsDao and then call that method in `getMatchedAds` method.